RecoJets/JetPlusTracks/python/PATJetPlusTrackCorrections_cff.py

```diff
@@ -0,0 +1,93 @@
+                                       jets = "slimmedCaloJets",
+                                       tracks = "produceTracks"
+)
+ak4JetTracksAssociatorAtCaloFaceJPTPAT = ak4JetTracksAssociatorAtCaloFace.clone(
+                                         jets = "slimmedCaloJets",
+                                         tracks = "produceTracks",
+                                         extrapolations = "trackExtrapolatorPAT"
+)
+ak4JetExtenderJPTPAT = ak4JetExtender.clone(
+                       jets = "slimmedCaloJets",
+                       jet2TracksAtCALO = "ak4JetTracksAssociatorAtCaloFaceJPTPAT",
+                       jet2TracksAtVX = "ak4JetTracksAssociatorAtVertexJPTPAT"
+)
+
+# ---------- Supported Modules
+
+from CommonTools.RecoAlgos.trackFromPackedCandidateProducer_cfi import *
+produceTracks = trackFromPackedCandidateProducer.clone()
+
+
+from RecoJets.JetPlusTracks.jetPlusTrackAddonSeedProducer_cfi import *
+JetPlusTrackAddonSeedRecoPAT = jetPlusTrackAddonSeedProducer.clone(
+    srcCaloJets = cms.InputTag("slimmedCaloJets"),
+    srcPVs = cms.InputTag('offlineSlimmedPrimaryVertices'),
+    UsePAT = True
+)
+
+
+from CommonTools.RecoAlgos.TrackWithVertexRefSelector_cfi import *
+trackWithVertexRefSelector.vertexTag = cms.InputTag('offlineSlimmedPrimaryVertices')
+trackWithVertexRefSelector.src = cms.InputTag('produceTracks')
+from RecoJets.JetProducers.TracksForJets_cff import *
+from RecoJets.Configuration.RecoTrackJets_cff import *
+ak4TrackJets.srcPVs = cms.InputTag('offlineSlimmedPrimaryVertices')
+ak4TrackJets.UseOnlyOnePV = cms.bool(True)
+
+# ---------- Module definition
+
+from RecoJets.JetPlusTracks.JetPlusTrackCorrections_cfi import *
+
+JetPlusTrackZSPCorJetAntiKt4PAT = cms.EDProducer(
+    "JetPlusTrackProducer",
+    cms.PSet(JPTZSPCorrectorAntiKt4),
+    src = cms.InputTag("slimmedCaloJets"),
+    srcTrackJets = cms.InputTag("ak4TrackJets"), 
+    srcAddCaloJets = cms.InputTag('JetPlusTrackAddonSeedRecoPAT'),
+    extrapolations = cms.InputTag("trackExtrapolatorPAT"),
+    tagName = cms.vstring('ZSP_CMSSW390_Akt_05_PU0'),
+    tagNameOffset = cms.vstring(),
+    PU = cms.int32(-1),
+    FixedPU = cms.int32(0),
+    UseZSP = cms.bool(False),
+    srcPVs = cms.InputTag('offlineSlimmedPrimaryVertices'),    
+    alias = cms.untracked.string('JetPlusTrackZSPCorJetAntiKt4'),
+    ptCUT = cms.double(15.),
+    dRcone = cms.double(0.4)
+    )
+
+JetPlusTrackZSPCorJetAntiKt4PAT.JetTracksAssociationAtVertex = "ak4JetTracksAssociatorAtVertexJPTPAT"
+JetPlusTrackZSPCorJetAntiKt4PAT.JetTracksAssociationAtCaloFace = "ak4JetTracksAssociatorAtCaloFaceJPTPAT"
+JetPlusTrackZSPCorJetAntiKt4PAT.Muons = "slimmedMuons"
+JetPlusTrackZSPCorJetAntiKt4PAT.Electrons = "slimmedElectrons"
+JetPlusTrackZSPCorJetAntiKt4PAT.JetSplitMerge = 2
+JetPlusTrackZSPCorJetAntiKt4PAT.UsePAT = True
+### ---------- Sequences
+
+# Anti-Kt
+
+PATJetPlusTrackCorrectionsAntiKt4 = cms.Sequence(
```

Comment:
cms.Task is preferred for scheduling module execution

please define a cms.Task (replace ```*``` with ```,``` in the module list)
```suggestion
PATJetPlusTrackCorrectionsAntiKt4Task = cms.Task(
```

and then, if the cms.Sequence is still needed, simply instantiate it from the task  e.g.
```PATJetPlusTrackCorrectionsAntiKt4 = cms.Sequence(PATJetPlusTrackCorrectionsAntiKt4Task)```

Reply:
Hi Slava, I do not why but Task does not work for PATxxx.cff. Moreover, this python module should not produce any output collection but just allow to work with JPT jets with MiniAOD on-f;ly.  With Tasks neither of modules are executed... It just reads the event and does nothing. 
All other item concerning this file I am including.

Reply:
Please clarify what is the problem.
We've made an equivalent update for many files in the past.